renderapi/transform/leaf/thin_plate_spline.py

```diff
@@ -280,3 +286,102 @@ def dataString(self):
         b64_2 = encodeBase64(blk2)
 
         return '{} {} {}'.format(header, b64_1, b64_2)
+
+    @staticmethod
+    def mesh_refine(
+            new_src,
+            old_src,
+            old_dst,
+            old_tf=None,
+            computeAffine=True,
+            tol=1.0,
+            max_iter=50,
+            nworst=10,
+            niter=0):
+
+        if old_tf is None:
+            old_tf = ThinPlateSplineTransform()
+            old_tf.estimate(old_src, old_dst, computeAffine=computeAffine)
+
+        new_tf = ThinPlateSplineTransform()
+        new_tf.estimate(
+                new_src,
+                old_tf.tform(new_src),
+                computeAffine=computeAffine)
+        new_dst = new_tf.tform(old_src)
+
+        delta = np.linalg.norm(new_dst - old_dst, axis=1)
+        ind = np.argwhere(delta > tol).flatten()
+
+        if ind.size == 0:
+            return new_tf
+
+        if niter == max_iter:
+            raise EstimationError(
```

Comment:
I think we want to define a custom EstimationError Exception that takes a message and a transform as an optional argument.  Then here you initialize the exception with a message and the tranformation that it got to.  The user can then decide to access that transform as a property of the Exception... or not. 
https://docs.python.org/3.3/tutorial/errors.html#tut-userexceptions